libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -517,6 +517,130 @@ string YulUtilFunctions::arrayLengthFunction(ArrayType const& _type)
 	});
 }
 
+std::string YulUtilFunctions::resizeDynamicArrayFunction(ArrayType const& _type)
+{
+	solAssert(_type.location() == DataLocation::Storage, "");
+	solAssert(_type.isDynamicallySized(), "");
+	solUnimplementedAssert(!_type.isByteArray(), "Byte Arrays not yet implemented!");
+	solUnimplementedAssert(_type.baseType()->isValueType(), "...");
+	solUnimplementedAssert(_type.baseType()->storageBytes() <= 32, "...");
+
+	if (!_type.isByteArray() && _type.baseType()->storageBytes() < 32)
+		solAssert(_type.baseType()->isValueType(), "Invalid storage size for non-value type.");
+
+	string functionName = "resize_array_" + _type.identifier();
+	return m_functionCollector->createFunction(functionName, [&]() {
+		return Whiskers(R"(
+			function <functionName>(array, newLen) {
+				let oldLen := <fetchLength>(array)
+
+				// Store new length
+				sstore(array, newLen)
+
+				// Size was reduced, clear end of array
+				if lt(newLen, oldLen) {
+					let oldSize := <convertToSize>(oldLen)
```

Comment:
Perhaps `oldSlotCount` to be more clear?

Reply:
hmm that would imply that it's only storage?

Reply:
Yes, for all other arrays you cannot change the length.